control/freqplot.py

```diff
@@ -1039,14 +1021,239 @@ def gangof4_plot(P, C, omega=None, **kwargs):
 
     plt.tight_layout()
 
+#
+# Singular value plot
+#
+
+
+# Default values for Bode plot configuration variables
+_singular_values_plot_default = {
+    'singular_values_plot.dB': False,           # Plot singular values in dB
+    'singular_values_plot.deg': True,           # Plot phase in degrees
+    'singular_values_plot.Hz': False,           # Plot frequency in Hertz
+    'singular_values_plot.grid': True,          # Turn on grid for gain and phase
```

Comment:
It seems a bit odd to me to have to set these separately from the equivalent `bode` configuration variable.  Is someone going to use one set of defaults for Bode and a separate one for singular values?  (Also note that `gang_of_four` uses the Bode plot defaults.)

If we don't want to use `bode.dB`, etc, we could also change all of them to `freqplot.dB` (though perhaps with some backward compatibility??).

Reply:
https://github.com/python-control/python-control/pull/593#discussion_r602856380

Reply:
I'd add the properties dB, Hz, and grid to freqplot and use these for bode, singular_values_plot, and gang_of_four.
The property _singular_values_plot_default would then disappear, and _bode_defaults would still contain bode.deg and bode.wrap_phase (and maybe unused dB, Hz, grid for backward compatibility, at least for the moment).

Is this OK?

Reply:
That seems fine. Make sure also to update the `use_matlab_defaults` and `use_fbs_defaults` functions in `config.py`.  I'm not sure about what to do with `bode.deg`.  On the one hand, it is true that this is the only function that plots phase, but at the same time seems odd to use `freqplot` for everything except that.  I would probably just use `freqplot` for everything.

Reply:
I pushed a new version where all the settings for bode_plot and singular_values_plot are in _freqplot_defaults

Reply:
I still see a `_singular_values_plot_default` with a comment for "Bode" above the definition and then used with_get_param('freqplot'). That's very confusing.

Also, personally, I would still prefer that we provide some backwards compatibility through an alias of config.defaults['bode'], so that old code does not break right away. Maybe including a warning message. I am not sure how to achieve this with a dict outside of `set_defaults` and `_get_param` in an elegant way.

Reply:
I tried to remove all references to the previous structure _singular_values_plot_default (new push).
Still have to handle backward compatibility in some way.